### PR TITLE
Creating and using Azure Function - Serverless Architecture in Azure portal and Visual Studio 2019

### LINKED ISSUE #3124: An overview of Azure Function - An Serverless Architecture
### Proposed title of the article

An overview of Azure Function - An Serverless Architecture

### Introduction paragraph (2-3 paragraphs):

An Azure Function is a perfect example of Serverless architecture or computes service. Using Azure Function, we can run the event-based code without managing any infrastructure. Since the Azure function is a trigger-based service, so it always executes a script of a block of code in response to a variety of events. An Azure Function can also be used to achieve the decoupling, high throughput or response, reusability of code, etc. So, we will discuss the Azure Functions in all aspects in four parts. 

### Key takeaways: 

1. What Azure Function entails
2. Azure Functions vs Web Jobs
3. Concept of ARM Templates and use of Azure Function.
4. Why we need to use Azure Function.
5. Creating Azure Functions with Help of Azure Portal.
6. Creating Azure Functions using Visual Studio 2019

### References: 
N/A

### PR DESCRIPTION
**Pull request by Dick7603**
complete article

### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

`This closes #3124 

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

File: content/articles/creating-and-using-azure-function-serverless-architecture-in-azure-portal-and-visual-studio-2019/index.md [hunk 1]
```diff
@@ -0,0 +1,216 @@
+### Introduction
+
+We may construct a serverless solution using Azure functions, which has the advantages of requiring less code, requiring less infrastructure upkeep, and cutting costs. So, despite putting a lot of effort into application deployments and server maintenance, cloud infrastructure like Azure provides us with all of the latest resources we need to keep our apps functioning.
+A fantastic example of Serverless architecture or computes service is an Azure Function. We can run event-based programs using Azure Function without having to manage any infrastructure. Because the Azure function is a trigger-based service, it always runs a script that contains a block of code in reaction to various events. Decoupling, fast throughput or responsiveness, reusability of code, and so on may all be accomplished with an Azure Function. Azure functions is a cloud-native design technique that allows you to deploy and run code without the requirement for server infrastructure, a web server, or any other setups.
+C#, Java, JavaScript, TypeScript, and Python are just a few of the languages that Azure Functions are available in.
```
Comment (jumamiller) [inline content/articles/creating-and-using-azure-function-serverless-architecture-in-azure-portal-and-visual-studio-2019/index.md:5]: Kindly breakdown the wordy intro.

File: content/articles/creating-and-using-azure-function-serverless-architecture-in-azure-portal-and-visual-studio-2019/index.md [hunk 2]
```diff
@@ -0,0 +1,216 @@
+### Introduction
+
+We may construct a serverless solution using Azure functions, which has the advantages of requiring less code, requiring less infrastructure upkeep, and cutting costs. So, despite putting a lot of effort into application deployments and server maintenance, cloud infrastructure like Azure provides us with all of the latest resources we need to keep our apps functioning.
+A fantastic example of Serverless architecture or computes service is an Azure Function. We can run event-based programs using Azure Function without having to manage any infrastructure. Because the Azure function is a trigger-based service, it always runs a script that contains a block of code in reaction to various events. Decoupling, fast throughput or responsiveness, reusability of code, and so on may all be accomplished with an Azure Function. Azure functions is a cloud-native design technique that allows you to deploy and run code without the requirement for server infrastructure, a web server, or any other setups.
+C#, Java, JavaScript, TypeScript, and Python are just a few of the languages that Azure Functions are available in.
+
+### Table of contents
+
```
Comment (jumamiller) [inline content/articles/creating-and-using-azure-function-serverless-architecture-in-azure-portal-and-visual-studio-2019/index.md:8]: Remove all these spaces between the heading and first line thoughout your article.

File: content/articles/creating-and-using-azure-function-serverless-architecture-in-azure-portal-and-visual-studio-2019/index.md [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+The Azure App Service Environment is a component of the Azure App Service that provides a fully isolated and dedicated environment for running App Service apps safely at scale.
+We can define the objects we desire, along with their types, names, and properties, in a JSON file that the ARM API can understand.
+The key benefit of using the ARM API is that it allows us to publish numerous resources in a single JSON file. We declare the categories of resources, the names we intend to use for those resources, and the attributes of the resources in this file. The ARM API then assigns the indicated properties to those resources, either by creating new objects that match these details or by changing an existing object if the resource name and type are the same.
+The ARM API adds resources to Azure, but it doesn't add any code or applications on those resources. For example, we can deploy a virtual machine with SQL Server already installed using an ARM template, but we can't deploy or restore a database using an SSDT package using an ARM template.
+
```
Comment (jumamiller) [inline content/articles/creating-and-using-azure-function-serverless-architecture-in-azure-portal-and-visual-studio-2019/index.md:61]: This one here is too wordy, kindly revise your article.

File: content/articles/creating-and-using-azure-function-serverless-architecture-in-azure-portal-and-visual-studio-2019/index.md [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+Our Function App is now available. Within this Function App, we must develop our first Azure Function.
+
+#### Creating Azure Function using Azure Portal
+
+**Step 1b**:
+
+Now, under the Function App area, select the Function choices from the left-side menu panel.
+
+![Step 1b](engineering-education-creating-and-using-azure-function-step-1b.png)
+
```
Comment (jumamiller) [inline content/articles/creating-and-using-azure-function-serverless-architecture-in-azure-portal-and-visual-studio-2019/index.md:147]: Fix all the images links, format is as follows: /engineering-education/folder_name/image_name.

**Review by jumamiller — CHANGES_REQUESTED** [review]
@Dick7603, I've suggested you fix your article before we move forward with the review process. It has very very long paragraphs which do away with the article flow.

**Comment by hectorkambow** [comment]
[Creating and using Azure Function - Serverless Architecture in Azure portal and Visual Studio 2019 #3653.pdf](https://github.com/section-engineering-education/engineering-education/files/7117917/Creating.and.using.Azure.Function.-.Serverless.Architecture.in.Azure.portal.and.Visual.Studio.2019.3653.pdf)


Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was higher than we typically accept.
Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content.
Be sure to see our resources page to see more info on plagiarism and what is considered as such.


@Dick7603

**Comment by Dick7603** [comment]
hello,
@hectorkambow  I have fixed the issues on plagiarism.
@jumamiller I have corrected the changes suggested.
Thank you.

**Comment by Muchiri021** [comment]
Hello, @hectorkambow @jumamiller I wish to enquire about the progress of my article. Thank you.

**Comment by Dick7603** [comment]
Hello, @hectorkambow @jumamiller I wish to enquire about the progress of my article. Thank you.

**Comment by jumamiller** [comment]
Hi @Dick7603 , @hectorkambow  will be running plagiarism tests soon. cc @ahmadmardeni1 @WanjaMIKE

**Comment by jumamiller** [comment]
@Dick7603 , this is an amazing article, however it  doesn't meet our set standards due to lack of uniqueness.

For us to proceed with the review, kindly rewrite this article, based on the title given, teach readers from basics of Azure functions. Make it as unique as possible. So far,everything on this article,  've already been covered on  [this](https://www.c-sharpcorner.com/article/azure-function-an-serverless-architecture/) article and [official docs](https://docs.microsoft.com/en-us/azure/azure-functions/functions-create-your-first-function-visual-studio?tabs=in-process) and even on [this](https://faun.pub/building-azure-functions-in-azure-portal-visual-studio-2017-using-net-core-7a598327a3d4) article.

cc @hectorkambow @louisefindlay23

Thanks!

**Comment by gracy20** [comment]
> @Dick7603 , this is an amazing article, however it doesn't meet our set standards due to lack of uniqueness.
> 
> For us to proceed with the review, kindly rewrite this article, based on the title given, teach readers from basics of Azure functions. Make it as unique as possible. So far,everything on this article, 've already been covered on [this](https://www.c-sharpcorner.com/article/azure-function-an-serverless-architecture/) article and [official docs](https://docs.microsoft.com/en-us/azure/azure-functions/functions-create-your-first-function-visual-studio?tabs=in-process) and even on [this](https://faun.pub/building-azure-functions-in-azure-portal-visual-studio-2017-using-net-core-7a598327a3d4) article.
> 
> cc @hectorkambow @louisefindlay23
> 
> Thanks!

Noted and I will be working on it. Thank you.

**Comment by Dick7603** [comment]
> @Dick7603 , this is an amazing article, however it doesn't meet our set standards due to lack of uniqueness.
> 
> For us to proceed with the review, kindly rewrite this article, based on the title given, teach readers from basics of Azure functions. Make it as unique as possible. So far,everything on this article, 've already been covered on [this](https://www.c-sharpcorner.com/article/azure-function-an-serverless-architecture/) article and [official docs](https://docs.microsoft.com/en-us/azure/azure-functions/functions-create-your-first-function-visual-studio?tabs=in-process) and even on [this](https://faun.pub/building-azure-functions-in-azure-portal-visual-studio-2017-using-net-core-7a598327a3d4) article.
> 
> cc @hectorkambow @louisefindlay23
> 
> Thanks!

Noted and I will be working on it. Thank you.

**Comment by LinusMuema** [comment]
Hi @Dick7603 
Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was higher than we typically accept. Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content. Be sure to see our resources page to see more info on plagiarism and what is considered as such.

[Creating and using Azure Function - Serverless Architecture in Azure portal and Visual Studio 2019 #3653 Copyleaks Report.pdf](https://github.com/section-engineering-education/engineering-education/files/7267392/Creating.and.using.Azure.Function.-.Serverless.Architecture.in.Azure.portal.and.Visual.Studio.2019.3653.Copyleaks.Report.pdf)

**Comment by Dick7603** [comment]
Hello, @LinusMuema @jumamiller I have fixed the issue. Thank you.
